src/modules/style.ts

```diff
@@ -1,9 +1,13 @@
 import {VNode, VNodeData} from '../vnode';
 import {Module} from './module';
 
-export type VNodeStyle = Record<string, string> & {
-  delayed?: Record<string, string>
-  remove?: Record<string, string>
+export type VNodeStyle = {
+  delayed?: { [prop: string]: string };
+  remove?: { [prop: string]: string };
+  destroy?: { [prop: string]: string };
+  [prop: string]: string | {
+    [prop: string]: string
+  } | undefined
```

Comment:
I tried this too, the problem with that variant is that it allows any properties also:
![image](https://user-images.githubusercontent.com/736697/29175847-98c5f988-7e03-11e7-9463-52b9e03d7214.png)